front/lib/resources/group_resource.ts

```diff
@@ -116,37 +123,58 @@ export class GroupResource extends BaseResource<GroupModel> {
     return new this(this.model, blob.get());
   }
 
-  static async fetchWorkspaceGroups(
-    auth: Authenticator,
+  static async superAdminFetchWorkspaceGroups(
+    userResource: UserResource,
+    workspaceId: ModelId,
     transaction?: Transaction
   ): Promise<GroupResource[]> {
-    const owner = auth.getNonNullableWorkspace();
+    if (!userResource.isDustSuperUser) {
+      throw new Error("User is not a super admin.");
+    }
 
     const groups = await this.model.findAll({
       where: {
-        workspaceId: owner.id,
+        workspaceId,
       },
       transaction,
     });
 
     return groups.map((group) => new this(GroupModel, group.get()));
   }
 
-  static async fetchWorkspaceSystemGroup(
-    auth: Authenticator,
+  static async fetchWorkspaceGroupsFromKey(
+    key: KeyResource,
+    transaction?: Transaction
+  ): Promise<GroupResource[]> {
+    const group = await this.model.findOne({
```

Comment:
Let's leave a `// TODO(GROUPS_INFRA)` saying that we need to pull the groups associated with the key once that's built